### PR TITLE
propagate dependencies support propagate pvc

### LINKED ISSUE #1887: Propagate dependencies add support propagate pvc and service
**What would you like to be added**:
Propagate dependencies add support propagate `pvc` and `service,` and give a option to user configure which resources need to dependend propagate
**Why is this needed**:
Propagate dependencies support propagate `configmap` and `secret` #1684

### PR DESCRIPTION
**Pull request by hanweisen**
Signed-off-by: hanweisen <hanweisen_yewu@cmss.chinamobile.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
propagate dependencies support propagate pvc
**Which issue(s) this PR fixes**:
Fixes #1887

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/persistentvolumeclaim.go
Comment (Tingtal) [inline test/e2e/framework/persistentvolumeclaim.go:76]:
```suggestion
// WaitPVCDisappearOnClusters: Wait for the service to disappear on member clusters until timeout.
```

Not sure whether my suggestion makes sense. It looks like a function description to me. Another thing is, why do we say "wait **service** disappear" in the description while the function name is "Wait**PVC**"? Shouldn't we say "Wait for the **PVC** to disappear"?

File: test/e2e/framework/persistentvolumeclaim.go
Comment (Tingtal) [inline test/e2e/framework/persistentvolumeclaim.go:78]:
```suggestion
	ginkgo.By(fmt.Sprintf("Check if pvc(%s/%s) diappears on member clusters", namespace, name), func() {
```

File: test/helper/resource.go
Comment (Tingtal) [inline test/helper/resource.go:592]:
```suggestion
// NewPVC will build a new PersistentVolumeClaim.
```

**Review by Tingtal — CHANGES_REQUESTED** [review]
Small touches on the language. Pls check whether my suggestions make sense. Thanks.

File: test/e2e/framework/persistentvolumeclaim.go
Comment (hanweisen) [reply test/e2e/framework/persistentvolumeclaim.go:76]: Yes, that's an error, `wait for the PVC to disappear` is right

Comment (hanweisen) [reply test/e2e/framework/persistentvolumeclaim.go]: I have to use `WaitPVCDisappearOnClusters` instead of `WaitPVCDisappearOnClusters:`, because use `WaitPVCDisappearOnClusters:` my goland will give me a warning, may this is a convention in `go`

File: pkg/resourceinterpreter/defaultinterpreter/dependencies.go
Comment (mrlihanbo) [inline pkg/resourceinterpreter/defaultinterpreter/dependencies.go:120]: dependentPVCs may be better?

**Comment by mrlihanbo** [comment]
Generally looks good to me. We need to clarify that the PVC will be propagated or migrated automatically，but the data in PV won't be migrated automatically. In other worlds, we just create a new pv in new clusters.

File: pkg/resourceinterpreter/defaultinterpreter/dependencies.go
Comment (hanweisen) [reply pkg/resourceinterpreter/defaultinterpreter/dependencies.go]: yes

**Comment by hanweisen** [comment]
> We need to clarify that the PVC will be propagated or migrated automatically，but the data in PV won't be migrated automatically. In other worlds, we just create a new pv in new clusters.
I think I can add this in user guide, after this pr is merged

**Review by RainbowMango — APPROVED** [review]
/approve
/hold
for @XiShanYongYe-Chang confirm

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/hold cancel

**Pull request merged**
